### PR TITLE
Add end_date to manual filter and region overrides parsing

### PR DESCRIPTION
**Pull request by TomGoBravo**
This PR addresses https://trello.com/c/weIWXUgV/1353-add-enddate-to-backend-metric-blocking

* Support `end_date` in region overrides JSON and the manual filter with pretty good unit test coverage
* Add a little more checking of date structure in check_timeseries_wide_dates_structure

### CODE REVIEW

**Comment by TomGoBravo** [comment]
[Add end_date to backend metric blocking.](https://trello.com/c/weIWXUgV/1353-add-enddate-to-backend-metric-blocking)

File: libs/datasets/manual_filter.py [hunk 1]
```diff
@@ -44,22 +45,33 @@ class Filter(pydantic.BaseModel):
...
                 )
-            if values["start_date"]:
+            if start_date:
                 raise ValueError(
                     "Filter including start_date without dropping observations "
                     "doesn't make sense."
                 )
+            if end_date:
+                raise ValueError(
+                    "Filter including end_date without dropping observations " "doesn't make sense."
```
Comment (mikelehen) [inline libs/datasets/manual_filter.py:70]: black reformatted your string funky

File: tests/libs/datasets/manual_filter_test.py [hunk 2]
```diff
@@ -313,6 +316,108 @@ def test_region_overrides_transform_and_filter_infection_rate():
...
+    "start_date, end_date, result, tag_date",
+    [
+        ("2020-04-03", None, [1, 2, None, None], "2020-04-03"),
+        (None, "2020-04-02", [None, None, 3, 4], "2020-04-02"),
+        ("2020-04-02", "2020-04-03", [1, None, None, 4], "2020-04-02"),
+        ("2020-04-02", "2020-04-02", [1, None, 3, 4], "2020-04-02"),
+        ("2020-04-05", None, [1, 2, 3, 4], None),
+        (None, "2020-03-28", [1, 2, 3, 4], None),
+    ],
+)
```
Comment (mikelehen) [inline tests/libs/datasets/manual_filter_test.py:329]: Nice!

**Review by mikelehen — APPROVED** [review]
Looks good!

**Pull request merged**